app/models/miq_request_task/state_machine.rb

```diff
@@ -68,9 +68,15 @@ def prematurely_finished?
     false
   end
 
-  def requeue_phase
+  # Retry up to 10k times by default which is roughly for 27h if we repeat every 10s
+  def requeue_phase(max_retries: 10**4)
```

Comment:
10,000 retries? That seems like a lot.

Reply:
Yeah so currently there is no limit to number of retries and I have no idea what kind of nasty lengthy stuff is being run by internal state machines, hence I wanted to put something really high. Bear in mind each retry generally happens after 10 seconds, that's why the number is high.

For the usecase I need this (physical server provisioning) I would set it to say 3 hours (about 1000 retries), but I'm not sure all other state machines are happy with 3 hours.

So what do you suggest? :)

Reply:
@miha-plesko Good question, I just wanted to make sure that was deliberate. I will defer to @gmcculloug and others who are more familiar with it.

Reply:
BTW I think Task currently appends to it's list of "execution servers" (which we check the length here) for each retry. In case of a bad condition when polling never stops, we actually append 1k, 10k, 100k, ... "execution servers" which means we're consuming more and more VMDB storage every 10 seconds. Or I'm missing something?

Reply:
Honestly it's probably not enough.  There are processes that can take multiple days.